### DISCUSSION TITLE
Sonoff POW R2: how to get actual values for fields on StatusPTH (PowerLow, PowerHigh, VoltageLow, VoltageHigh, CurrentLow, CurrentHigh)?

### DISCUSSION THREAD

**Discussion by DurvalMenezes** [arendst/Tasmota#18156] [Q&A] [answered]
As per the subject. These fields always show as zero when queried via "Status 9":

```
$ curl --silent "http://gen03/cm?user=XXX&password=XXX&cmnd=Status%209"; echo

{"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
```

The sensor data (current power, voltage, current etc) are being read ok as shown by "Status 10":

```
curl --silent "http://gen03/cm?user=XXX&password=XXX&cmnd=Status%2010"; echo

{"StatusSNS":{"Time":"2023-03-10T17:42:17","ENERGY":{"TotalStartTime":"2022-11-24T23:56:43","Total":56.066,"Yesterday":0.396,"Today":2.029,"Power":711,"ApparentPower":711,"ReactivePower":0,"Factor":1.00,"Voltage":235,"Current":3.026}}}
```

Do I have to do anything like setting a measurement period or whatever in order to get these values to actually record what they're supposed to?

In case it makes a difference, I'm running Tasmota 12.2.0.

TIA!

**Comment by barbudor** [comment, marked as answer]
> 9 = show power thresholds (only on modules with power monitoring)

This PTH status is showing the value of threshold settings not actual measurments
It summarizes the values set with commands of the same name
So it's actually doing exactly what it's supposed to do

**Reply by DurvalMenezes** [reply]
Hummrmm... ok.

Is there a way to set Tasmota to record low/high values for these, and show them when requested?

**Reply by sfromis** [reply]
No. You can to some extent create rules to react to values exceeding the previously recorded one, but that does not get them in the web UI (which is not meant for daily operation anyway). Or do it in your backend for whole TelePeriod intervals.

**Reply by DurvalMenezes** [reply]
Thanks for the clarification. I'm setting rules for some limits I want to monitor both for VoltageHigh and PowerHigh, my idea is to log something to my syslog server when these values are exceeded.

Any tips on how to do this?

**Reply by sfromis** [reply]
Here's a cookbook example illustrating checking values against prev, and doing publish. Details have to be adapted.
https://tasmota.github.io/docs/Rules/#transmit-sensor-value-only-when-a-delta-is-reached

Note that if it "only" is a question of exceeding thresholds, Tasmota already publishes messages when that happens.
